src/room.ts

```diff
@@ -209,21 +230,40 @@ export class Room extends EventEmitter implements Sayable {
     return rawMemberList.map(m => Contact.load(m.UserName))
   }
 
-  private parseNickMap(memberList: RoomRawMember[]): Map<string, string> {
-    const nickMap: Map<string, string> = new Map<string, string>()
-
+  /**
+   * ISSUE #64 emoji need to be striped
+   * ISSUE #104 never use remark name because sys group message will never use that
+   * @rui: cannot use argument NickName because it mix real nick and remark 
+   */
+  private parseMap(memberList: RoomRawMember[], parseContent: string): Map<string, string> {
+    const mapList: Map<string, string> = new Map<string, string>()
     if (memberList && memberList.map) {
       memberList.forEach(member => {
-        /**
-         * ISSUE #64 emoji need to be striped
-         * ISSUE #104 never use remark name because sys group message will never use that
-         */
-        nickMap[member.UserName] = UtilLib.stripEmoji(
-          member.DisplayName || member.NickName
-        )
+        let contact = Contact.load(member.UserName)
+        if (parseContent === 'nick') {
+          mapList[member.UserName] = UtilLib.stripEmoji(contact.name())
+        } else if (parseContent === 'remark') {
+          mapList[member.UserName] = UtilLib.stripEmoji(contact.remark() || '')
+        } else if (parseContent === 'display') {
+          mapList[member.UserName] = UtilLib.stripEmoji(member.DisplayName)
+        } else {
+          throw new Error('contact not parseContent')
+        }
       })
     }
-    return nickMap
+    return mapList
+  }
+
+  private parseNickMap(memberList: RoomRawMember[]): Map<string, string> {
+    return this.parseMap(memberList, 'nick')
+  }
+
+  private parseRemarkMap(memberList: RoomRawMember[]): Map<string, string> {
+    return this.parseMap(memberList, 'remark')
+  }
+
+  private parseDisplayName(memberList: RoomRawMember[]): Map<string, string> {
+    return this.parseMap(memberList, 'display')
```

Comment:
Get rid of unnessesary functions: parseNickMap / parseRemarkMap / parseDisplayName

use `this.parseMap(memberList, 'display')` directly